src/nix/flake.cc

```diff
@@ -146,14 +174,11 @@ struct CmdFlakeUpdate : StoreCommand, GitRepoCommand, MixEvalArgs
 
     void run(nix::ref<nix::Store> store) override
     {
-        auto evalState = std::make_shared<EvalState>(searchPath, store);
-
-        if (gitPath == "") gitPath = absPath(".");
-        updateLockFile(*evalState, gitPath);
+        getFlake();
```

Comment:
1. We now have a third `getFlake` function, which has different behavior from the other two. This will make it very hard to hack on the Nix codebase.
2. I think `nix flake update` should show an error if it is called on a FlakeRef which isn't of the form `IsPath`.

Reply:
I've restored the old behaviour for `nix flake update`.